core/src/main/java/com/datastax/oss/driver/internal/core/graal/DefaultFeature.java

```diff
@@ -0,0 +1,122 @@
+ */
+@AutomaticFeature
+final class DefaultFeature implements Feature {
+
+  public Config buildConfig() {
+
+    /* Copied from DefaultDriverConfigLoader, including the DEFAULT_ROOT_PATH value there as the literal
+    "datastax-java-driver" below.  Can't reference the class directly here since that will introduce
+    a requirement to specify it as a build-time dependency (as well as the slf4j logger it creates
+    via static init */
+    return ConfigFactory.defaultOverrides()
+        .withFallback(ConfigFactory.defaultApplication())
+        .withFallback(ConfigFactory.defaultReference(CqlSession.class.getClassLoader()))
+        .resolve()
+        .getConfig("datastax-java-driver");
+  }
+
+  public Optional<String> tryString(Config config, DefaultDriverOption option) {
+
+    try {
+      return Optional.of(config.getString(option.getPath()));
+    } catch (ConfigException.WrongType e) {
+      return Optional.empty();
+    } catch (ConfigException.Missing e) {
+      return Optional.empty();
+    }
+  }
+
+  public Optional<Class<?>> tryLoadClass(String configVal, String... defaultPackages) {
+
+    try {
+      if (configVal.contains(".")) {
+        return Optional.of(Class.forName(configVal));
+      } else {
+        for (String defaultPackage : defaultPackages) {
+          String qualifiedClassName = defaultPackage + "." + configVal;
+          try {
+            return Optional.of(Class.forName(qualifiedClassName));
+          } catch (ClassNotFoundException cnfe) {
+          }
+        }
+        return Optional.empty();
+      }
+    } catch (LinkageError | Exception e) {
+      return Optional.empty();
+    }
+  }
+
+  public void tryRegister(Config config, DefaultDriverOption option, String... defaultPackages) {
+
+    tryString(config, option)
+        .flatMap(configVal -> tryLoadClass(configVal, defaultPackages))
+        .ifPresent(
+            (clz) -> {
+              /* Mark the class as being initialized at build time */
+              RuntimeClassInitialization.initializeAtBuildTime(clz);
+
+              /* Register all the things as being available at runtime */
+              RuntimeReflection.register(clz);
+              for (Method m : clz.getMethods()) {
+                RuntimeReflection.register(m);
+              }
+              for (Field f : clz.getFields()) {
+                RuntimeReflection.register(f);
+              }
+              for (Constructor c : clz.getConstructors()) {
+                RuntimeReflection.register(c);
+              }
+            });
+  }
+
+  @Override
+  public void beforeAnalysis(Feature.BeforeAnalysisAccess access) {
+
+    /* Make the Typesafe classes we need to do our work available at build-time */
+    RuntimeClassInitialization.initializeAtBuildTime("com.typesafe.config.impl");
+
+    /* slf4j logger class, used by the parent class of the default LBP (and probably plenty more) */
+    RuntimeClassInitialization.initializeAtBuildTime("org.slf4j.LoggerFactory");
```

Comment:
We don't have to actually instantiate the classes but we do wind up loading them here, so static initializers come into play.  Superclasses must also be resolved, so in this case LoggerFactory is actually used in BasicLoadBalancingPolicy, the superclass of DefaultLoadBalancingPolicy.

Like the Typesafe packages above the goal is to ultimately replace this with something more dynamic; this is just a placeholder to keep things moving for now.

Reply:
Ouch. I naively thought that once you register some class for reflection, imported classes would be automatically detected.